### PR TITLE
feat: port MMI related logic to personal sign page

### LINKED ISSUE #22970: Personal Sign Redesign - port current MMI related logic to new designs
There is MMI code that is executed a signature is approved / rejected etc. We need to port it to new deisgns.

### PR DESCRIPTION
**Pull request by jpuri**
## **Description**

Port current personal sign MMI related logic to new designs.

## **Related issues**

Fixes: MetaMask/metamask-extension#22970

## **Manual testing steps**
NA

## **Screenshots/Recordings**
NA

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've clearly explained what problem this PR is solving and how it is solved.
- [X] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [X] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [X] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by jpuri** [comment]
We would need help from MMI team to review + QA the changes.

File: ui/pages/confirmations/utils/confirm.test.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { ApprovalRequest } from '@metamask/approval-controller';
+import { ApprovalType } from '@metamask/controller-utils';
+
+import { isSignatureApprovalRequest } from './confirm';
+
+describe('confirm util', () => {
+  describe('isSignatureApprovalRequest', () => {
+    it('should return true for signature approval requests', () => {
```
Comment (segun) [inline ui/pages/confirmations/utils/confirm.test.ts:8]:
```suggestion
    it('returns true for signature approval requests', () => {
```

File: ui/pages/confirmations/utils/confirm.test.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      const result = isSignatureApprovalRequest({
+        type: ApprovalType.PersonalSign,
+      } as ApprovalRequest<any>);
+      expect(result).toStrictEqual(true);
+    });
+    it('should return false for request not of type signature', () => {
```
Comment (segun) [inline ui/pages/confirmations/utils/confirm.test.ts:14]:
```suggestion
    it('returns false for request not of type signature', () => {
```

**Review by segun — DISMISSED** [review]
LGTM asides a few comments on test descrption

File: ui/pages/confirmations/components/confirm/footer/footer.tsx [hunk 3]
```diff
@@ -8,16 +8,34 @@ import {
...
   const currentConfirmation = useSelector(currentConfirmationSelector);
+  ///: BEGIN:ONLY_INCLUDE_IF(build-mmi)
+  const { mmiOnSignCallback, mmiSubmitDisabled } = useMMIConfirmationInfo();
+  ///: END:ONLY_INCLUDE_IF
+
+  let from: string | undefined;
+  // todo: extend to other confirmation types
+  if (currentConfirmation?.msgParams) {
+    from = currentConfirmation?.msgParams?.from;
+  }
```
Comment (pedronfigueiredo) [inline ui/pages/confirmations/components/confirm/footer/footer.tsx:31]:
```suggestion
const from: string | undefined = currentConfirmation?.msgParams?.from;
```

Comment (jpuri) [reply ui/pages/confirmations/components/confirm/footer/footer.tsx:31]: Hey @pedronfigueiredo : I wrote it like this as other transaction type like send, approve etc will have different logic to initialise `from`.

**Review by pedronfigueiredo — DISMISSED** [review]

**Review by pedronfigueiredo — APPROVED** [review]

**Review by segun — APPROVED** [review]

**Pull request merged**